controllers/topology/reconcile_state.go

```diff
@@ -56,6 +58,69 @@ func (r *ClusterReconciler) reconcileInfrastructureCluster(ctx context.Context,
 	return r.reconcileReferencedObject(ctx, current.infrastructureCluster, desired.infrastructureCluster)
 }
 
+// reconcileControlPlane works to bring the current state of a managed topology in line with the desired state. This involves
+// updating the cluster where needed.
+func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, current, desired *clusterTopologyState) error {
+	log := ctrl.LoggerFrom(ctx)
+	// Set a default nil return function for the cleanup operation.
+	cleanup := func() error { return nil }
+
+	// If the ControlPlane requires an InfrastructureMachineTemplate
+	// TODO: Check to ensure this is in the clusterclass - make this a common function.
+	// There's a possible issue with using get here - getControlPlaneState uses a class passed to it.
+	// This function also adds a side effect as it's a network call to an outside component - makes testing much more difficult.
```

Comment:
As pointed out by @sbueringer above, this could result in getting a different version of the class than what the desired state is based off of because it's another API server query at another point in time. This changes the reconcile from an isolated operation to one that has some possibly mutating dependency underneath it.

Reply:
I agree if we are going to read the class again, but this should not happen because all the information are already there, right?

Reply:
I don't think the class information is in either the desired or current state - so I think here it would have to be passed down the function chain in order to get access to it. Am I missing somewhere we can get access to the class?

Reply:
If we need the class, we should pass it down. That's fine.

Reply:
I've made changes to bring the class down the full call chain, but I think this needs further discussion.